tendrl/node_agent/monitoring/collectd/collectors/gluster/low_weight/tendrl_glusterfs_brick_disk_stats.py

```diff
@@ -0,0 +1,421 @@
+        out, err = gluster_utils.exec_command(command)
+        if err:
+            return None, None, err
+        mount_source, mount_point = out.split("\n")[-2].split()
+        return mount_source, mount_point, None
+
+    def get_brick_devices(self, brick_path):
+        mount_source, mount_point, err = self.get_brick_source_and_mount(
+            brick_path
+        )
+        if err or (not mount_source or not mount_point):
+            collectd.error(
+                'mount_source is %s\n mount_point is %s\n DEVICE_TREE is %s\n'
+                'Failed to fetch device details of brick %s.'
+                'Error %s.' % (
+                    str(mount_source),
+                    str(mount_point),
+                    str(self.DEVICE_TREE),
+                    brick_path,
+                    err
+                )
+            )
+            return None, None
+        device = self.DEVICE_TREE.resolveDevice(mount_source)
+        disks = [
+            d.path for d in device.ancestors if d.isDisk and not d.parents
+        ]
+        return disks, mount_point
+
+    def get_interval_disk_io_stat(self, device_name, attr_name):
+        return (
+            (
+                getattr(self.current_io_stats[
+                    device_name.replace(
+                        '/dev/',
+                        ''
+                    )
+                ], attr_name) -
+                getattr(self.initial_io_stats[
+                    device_name.replace(
+                        '/dev/',
+                        ''
+                    )
+                ], attr_name)
+            ) * 1.0
+        ) / (self.STAT_INTERVAL_FOR_PER_SEC_COUNTER * 1.0)
+
+    def get_disk_usage(self, device_name):
+        return psutil.disk_usage(device_name)
+
+    def populate_disk_details(self, vol_name, brick_host, brick_path):
+        try:
+            brick_devices, mount_point = self.get_brick_devices(brick_path)
+            if not brick_devices:
+                collectd.error(
+                    'Failed to fetch device details for brick %s:%s'
+                    ' of volume %s' % (
+                        brick_host,
+                        brick_path,
+                        vol_name
+                    )
+                )
+                return
+            for brick_device in brick_devices:
+                # Collect disk read and write octets
+                # Push to cluster->volume->node->brick tree
+                self.brick_details[
+                    'clusters.%s.volumes.%s.nodes.%s.bricks.%s.device.%s.'
+                    'disk_octets.read' % (
+                        self.CONFIG['integration_id'],
+                        vol_name,
+                        brick_host.replace('.', '_'),
+                        brick_path.replace('/', '|'),
+                        brick_device.replace('/dev/', '')
+                    )
+                ] = self.get_interval_disk_io_stat(
```

Comment:
@anmolbabu , why you are making these method calls again and again(same as in line 131). Instead you can make the call once use it at both places right? Is there specific reason why you are doing like this?

I think the same is applicable at multiple places in this method

Reply:
There's not much gain right..
I mean the info even now is fetched only once but diff calculations are happening on demand